### PR TITLE
Support binary response type #141

### LINKED ISSUE #141: Support binary response type
**Is your feature request related to a problem? Please describe.**
Laconia adapter-api is not supporting binary type like Buffer or Stream.

**Describe the solution you'd like**
Application core layer must be able to return Stream or Buffer. When Stream of Buffer is received, the response content type will be set to `application/octet-stream`.

```js
const adapterApi = require("@laconia/adapter-api");

const app = () => { /* return Buffer */ }
const apigateway = adapterApi.apigateway()

exports.handler = laconia(apigateway(app)).register(instances);
```

**Additional context**

See how micro handles this: [here](https://github.com/zeit/micro/blob/069f31fa506086cdbf0ce3364d16ab972e8cc200/lib/index.js#L28)
See how aws-serverless-express handles this: [here](https://github.com/awslabs/aws-serverless-express/blob/86c0c58009f26b57f319d9d5089fd1a1cf86ce06/src/index.js#L68)
See [this thread](https://stackoverflow.com/questions/41429551/aws-api-gateway-base64decode-produces-garbled-binary/47780921#47780921) on how AWS API Gateway is handling binary type

### PR DESCRIPTION
**Pull request by hugosenari**
Resolves #141 

### Problem:

To convert body from stream to base64 we need a Promise/Async/Callback, but convert ApiGatewayResponse.create to async would break api compatibility.

### Curiosity:

Why number is JSON.stringify(body) but 'text/plain', instead of 'application/json'?

### CODE REVIEW

File: packages/laconia-event/test/apigateway/ApiGatewayResponse.spec.js [hunk 1]
```diff
@@ -83,4 +84,74 @@ describe("ApiGatewayResponse", () => {
...
+      );
+    });
+
+    it("should set isBase64Encoded to true", async () => {
+      const response = await ApiGatewayResponse.create(output);
+      expect(response).toBeBase64Encoded(true);
+    });
+  });
+
+  describe("when convertin a Stream", () => {
```
Comment (ceilfors) [inline packages/laconia-event/test/apigateway/ApiGatewayResponse.spec.js]: Missing `g` in converting :)

**Comment by hugosenari** [comment]
> Is there a strong case for number to be application/json instead of text/plain?

Pro:

* Any system that auto convert response based in headers will convert it to Number

Con:

* Retro compatibility issues if someone assert that his random num generator returns text/plain and it changes.

File: packages/laconia-event/src/apigateway/getContentType.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
+const getResponseProps = require("./getResponseProps");
```
Comment (hugosenari) [inline packages/laconia-event/src/apigateway/getContentType.js]: To be honest I don't know if we need this file anymore.

The only reference we have now are tests.

**Comment by hugosenari** [comment]
test API https://mc5hzv2fok.execute-api.eu-west-1.amazonaws.com/node8/image

I couldn't deploy all laconia-accetance to create/run acceptance tests, any tip?

TODO:
- [ ] Update docs informing [correct configuration](https://serverless.com/framework/docs/providers/aws/events/apigateway/#binary-media-types);
- [ ] Write acceptance tests;

**Comment by ceilfors** [comment]
Hey @hugosenari, thanks for your patience, I just got back from my holiday. I've reviewed the change and it looks good to me!

I'm happy to merge the current code, and you can continue the TODO is other PR if you're keen (the acceptance test is known to be a pain :( )

**Pull request merged**